service/bfg/bfg.go

```diff
@@ -1198,16 +1275,35 @@ func (s *Server) handleBtcFinalityByKeystonesRequest(ctx context.Context, bfkr *
 	}, nil
 }
 
-func (s *Server) handleL2KeystonesRequest(ctx context.Context, l2kr *bfgapi.L2KeystonesRequest) (any, error) {
-	log.Tracef("handleL2KeystonesRequest")
-	defer log.Tracef("handleL2KeystonesRequest exit")
+func (s *Server) getL2KeystonesCache() []hemi.L2Keystone {
+	log.Tracef("getL2KeystonesCache")
+	defer log.Tracef("getL2KeystonesCache exit")
+
+	s.mtx.Lock()
+	defer s.mtx.Unlock()
+
+	if s.l2keystonesCache == nil {
+		return []hemi.L2Keystone{}
```

Comment:
Is there a reason to initialise this slice? If not, I think we should just return `nil`:
```suggestion
		return nil
```

Reply:
ah no, some fallout from previous iteration, will change